testsuite/pytests/test_spatial/test_basics.py

```diff
@@ -345,6 +345,54 @@ def test_GetTargetNodes(self):
         self.assertEqual(len(t), 1)
         self.assertEqual(t[0], nest.NodeCollection([1, 3, 7, 9]))
 
+    def test_GetSourceNodes(self):
+        """Interface check for finding source."""
+
+        cdict = {'rule': 'pairwise_bernoulli',
+                 'p': 1.,
+                 'mask': {'grid': {'shape': [2, 2]}}}
+        sdict = {'synapse_model': 'stdp_synapse'}
+        nest.ResetKernel()
+        # For co-dependent properties, we use `set()` instead of kernel attributes
+        nest.set(use_compressed_spikes=False, sort_connections_by_source=False)
+
+        layer = nest.Create('iaf_psc_alpha',
+                            positions=nest.spatial.grid(shape=[3, 3],
+                                                        extent=(2., 2.),
+                                                        edge_wrap=True))
+
+        # connect layer -> layer
+        nest.Connect(layer, layer, cdict, sdict)
+
+        s = nest.GetSourceNodes(layer, layer[0])
+        self.assertEqual(len(s), 1)
+
+        s = nest.GetSourceNodes(layer, layer)
+        self.assertEqual(len(s), len(layer))
+        # 2x2 mask -> each nodes has four sources
+        self.assertTrue(all([len(g) == 4 for g in s]))
+
+        s = nest.GetSourceNodes(layer, layer, syn_model='static_synapse')
+        self.assertEqual(len(s), len(layer))
+        self.assertTrue(all([len(g) == 0 for g in s]))  # no static syns
+
+        s = nest.GetSourceNodes(layer, layer, syn_model='stdp_synapse')
+        self.assertEqual(len(s), len(layer))
+        self.assertTrue(
+            all([len(g) == 4 for g in s]))  # 2x2 mask  -> four sources
```

Comment:
No need to break the line here.
```suggestion
        self.assertTrue(all([len(g) == 4 for g in s]))  # 2x2 mask  -> four sources
```